datadog-checks-base/datadog_checks/config.py

```diff
@@ -0,0 +1,23 @@
+# Unless explicitly stated otherwise all files in this repository are licensed
+# under the Apache License Version 2.0.
+# This product includes software developed at Datadog (https://www.datadoghq.com/).
+# Copyright 2018 Datadog, Inc.
+
+
+def is_affirmative(value):
+    """
+    Attempt to convert different type of values to a meaningful boolean
+    """
+    if value is None:
+        return False
+
+    # int or real bool
+    if isinstance(value, int):
+        return bool(value)
+
+    # try string cast
+    return value.lower() in ('yes', 'true', '1', 'y', 'on')
```

Comment:
I'd simplify this to:

```
def is_affirmative(value):
    """
    Attempt to convert different type of values to a meaningful boolean
    """
    # try string cast
    if isinstance(value, str):
        return value.lower() in {'yes', 'true', '1', 'y', 'on'}

    # use object's implementation of `__bool__`.
    # None -> False, 0 -> False, 1 -> True, etc.
    return not not value
```

I assume we call this function frequently, so for performance reasons I've changed the tuple to a set and removed the slow global `bool` lookup.

Reply:
This looks like a nice improvement. We don't normally call `is_affirmative` all that often, it's mostly called at start-up when parsing the configuration files. Still, this is still readable and more efficient.

Reply:
is `__bool__` actually called in this case? I suspect it'd be `__nonzero__`, same would happen when casting to bool with `bool(value)`. Do you know how big is the difference between `not not Value` and `bool(value)`? If it's not that much, I think the latter is more readable.

Reply:
You're right, Python 3 changed `__nonzero__` to `__bool__`. The only difference is it's no longer valid for users' implementations to return 0 or 1.

Built-in/stdlib types do the right thing and return bools:

```
Python 2.7.12 | packaged by conda-forge | (default, Sep  8 2016, 14:46:05) [MSC v.1500 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.                                         
>>> bool(None)                                                                                                 
False                                                                                                          
>>> not not None                                                                                               
False                                                                                                          
>>>                                                                                                            
>>> bool(0)                                                                                                    
False                                                                                                          
>>> not not 0                                                                                                  
False                                                                                                          
>>>                                                                                                            
>>> bool(1)                                                                                                    
True                                                                                                           
>>> not not 1                                                                                                  
True                                                                                                           
>>>                                                                                                            
>>> bool([])                                                                                                   
False                                                                                                          
>>> not not []                                                                                                 
False                                                                                                          
>>>                                                                                                            
>>> bool([0])                                                                                                  
True                                                                                                           
>>> not not [0]                                                                                                
True                                                                                                           
```

As far as performance:

```

```

Reply:
Cool, will keep `not not value` and update the comment, thanks!